query/sources/common/src/main/java/org/codice/ddf/admin/sources/utils/RequestUtils.java

```diff
@@ -217,6 +210,18 @@ private WebClient generateClient(String url, CredentialsField creds,
         return client;
     }
 
+    /**
+     * This is broken out for testing purposes so that the SecureCxfClientFactory can be mocked out.
+     */
+    protected SecureCxfClientFactory<WebClient> createFactory(String url, CredentialsField creds) {
+        return creds.username() == null && creds.password() == null ? new SecureCxfClientFactory<>(
```

Comment:
✏️ This would allow for a null username and non-null password to be passed to the factory ctor. I can't think that's something we would want to support.